velox/docs/functions/spark/string.rst

```diff
@@ -236,22 +236,26 @@ Unless specified otherwise, all functions return NULL if at least one of the arg
 
         SELECT soundex('Miller'); -- "M460"
 
-.. spark:function:: split(string, delimiter) -> array(string)
-
-    Splits ``string`` on ``delimiter`` and returns an array. ::
+.. spark:function:: split(string, delimiter[, limit]) -> array(string)
+
+    Splits ``string`` around occurrences that match ``delimiter`` and returns an array with a length of
+    at most ``limit``. ``delimiter`` is a string representing regular expression. ``limit`` is an integer
+    which controls the number of times the regex is applied. By default, ``limit`` is -1. When ``limit`` > 0,
+    the resulting array's length will not be more than ``limit``, and the resulting array's last entry will
+    contain all input beyond the last matched regex. When ``limit`` <= 0, ``regex`` will be applied as many
+    times as possible, and the resulting array can be of any size. When ``delimiter`` is empty, if ``limit``
+    is smaller than the size of ``string``, the resulting array only contains ``limit`` number of single characters
+    splitting from ``string``, if ``limit`` is not provided or is larger than the size of ``string``, the resulting 
+    array contains all the single characters of ``string`` and does not include an empty tail character.
+    The split function align with vanilla spark 3.4+ split function. ::
```

Comment:
> The split function align with vanilla spark 3.4+ split function.

Wondering why do we need this comment? Isn't it true for all functions that they are expected to match Spark 3.4+ behavior?

Reply:
just in case people see different behavior happen for spark version below 3.4 as 3.2/3.3

Reply:
What about other functions? Do they implement Spark 3.2/3.3 semantics? CC: @rui-mo

Reply:
@gaoyangxiaozhu @rui-mo gentle ping

Reply:
@mbasmanova For the split function, the behavior for empty delimiter was changed since Spark 3.4 with commit https://github.com/apache/spark/commit/247306c0fe8cfda17c4182ce2338f7312e3aa897.
When implementing functions in Velox, if there is semantic difference among Spark versions, I assume we need to follow the latest one. Does it makes sense? Thanks.

Reply:
@rui-mo Rui, thank you for clarifying. I assume this is a general policy that applies to all functions. If so, it would be helpful to document it in https://facebookincubator.github.io/velox/spark_functions.html and clarify which Spark version we match and whether we match ANSI mode on or off.

Furthermore, assuming Spark doesn't guarantee backwards compatibility across versions, we would have to pick a specific version we match and cannot say "latest" or 3.4+. If Spark changes behavior in "latest" version and we change to match, existing users will start seeing different behavior. Is this the case?

Reply:
@mbasmanova Thanks for the pointer. I'd like to document it.
>  If Spark changes behavior in "latest" version and we change to match, existing users will start seeing different behavior. Is this the case?

I assume it is the case, and it makes sense to me to pick a specific version.

Reply:
Opened https://github.com/facebookincubator/velox/pull/10677 for the documentation.

@gaoyangxiaozhu I assume below statement and `Since Spark 3.4` for `splitEmptyDelimiter` function could be removed.

> The split function align with vanilla spark 3.4+ split function.